File: gr.demokritos.iit.graphmethods/src/gr/demokritos/iit/graphmethods/BinaryGraphOperator.java
package gr.demokritos.iit.graphmethods;

import org.apache.spark.graphx.Graph;

/**
 *
 * @author JKLS
 */
public interface BinaryGraphOperator {
    
    /**
     * Creates a new graph based on an operation
     * @param g1 graph1
     * @param g2 graph2
     * @return graph after operation upon it
     */
    public Graph getResult(Graph g1, Graph g2);
    
}

File: GraphMethods/src/main/java/BinaryGraphOperator.scala
import org.apache.spark.graphx.Graph

/**
 * @author Kontopoulos Ioannis
 */
trait BinaryGraphOperator {

  //@return graph after specific operation
  def getResult(g1: Graph[String, Double], g2: Graph[String, Double]): Graph[String, Double]

}